packages/@aws-cdk/aws-iam/lib/oidc-provider.ts

```diff
@@ -129,6 +156,10 @@ export class OpenIdConnectProvider extends Resource implements IOpenIdConnectPro
       },
     });
 
+    this.openIdConnectProviderIssuer = props.url
```

Comment:
This is actually a bug because `props.url` can be a token, in which case the `replace` will simply not take affect.

Instead, we should use token operations:

```suggestion
    this.openIdConnectProviderIssuer = Fn.select(1, Fn.split('oidc-provider/', this.openIdConnectProviderArn));
```

Reply:
getting the
`{"Fn::Select": [1, {"Fn::Split": ["oidc-provider/", {"Ref": "MyProvider730BA1C8"}]}]}`

for this test:
`test('extract issuer properly in the new provider', () => {
    // GIVEN
    const stack = new Stack();

    // WHEN
    const provider = new iam.OpenIdConnectProvider(stack, 'MyProvider', {
      url: 'https://my-issuer',
    });

    // THEN
    expect(stack.resolve(provider.openIdConnectProviderIssuer)).toStrictEqual('MyProvider730BA1C8');
  });`

Reply:
I'm getting an error in one of the tests now:
`Expected: "MyProvider730BA1C8"
    Received: {"Fn::Select": [1, {"Fn::Split": ["oidc-provider/", {"Ref": "MyProvider730BA1C8"}]}]}`

here is the test:
```
test('extract issuer properly in the new provider', () => {
    // GIVEN
    const stack = new Stack();

    // WHEN
    const provider = new iam.OpenIdConnectProvider(stack, 'MyProvider', {
      url: 'https://my-issuer',
    });

    // THEN
    expect(stack.resolve(provider.openIdConnectProviderIssuer)).toStrictEqual('MyProvider730BA1C8');
  });
```
since the openIdConnectProviderArn has a value of `{ Ref: 'MyProvider730BA1C8' }` after I create a new one.
Not sure if it is correct.